mongo/integration/gridfs_test.go

```diff
@@ -29,6 +30,83 @@ func TestGridFS(x *testing.T) {
 	mt := mtest.New(x, noClientOpts)
 	defer mt.Close()
 
+	mt.Run("skipping download", func(mt *mtest.T) {
+		data := []byte("abc.def.ghi")
+		var chunkSize int32 = 4
+
+		testcases := []struct {
+			name string
+
+			read              int
+			skip              int64
+			expectedSkipN     int64
+			expectedSkipErr   error
+			expectedRemaining string
+		}{
+			{
+				"read 0, skip 0", 0, 0, 0, nil, "abc.def.ghi",
+			},
+			{
+				"read 0, skip to end of chunk", 0, 4, 4, nil, "def.ghi",
+			},
+			{
+				"read 0, skip 1", 0, 1, 1, nil, "bc.def.ghi",
+			},
+			{
+				"read 1, skip to end of chunk", 1, 3, 3, nil, "def.ghi",
+			},
+			{
+				"read all, skip beyond", 11, 1, 0, nil, "",
+			},
+			{
+				"skip all", 0, 11, 11, nil, "",
+			},
+			{
+				"read 1, skip to last chunk", 1, 8, 8, nil, "hi",
+			},
+			{
+				"read to last chunk, skip to end", 9, 2, 2, nil, "",
+			},
+			{
+				"read to last chunk, skip beyond", 9, 4, 2, nil, "",
+			},
+		}
+
+		for _, tc := range testcases {
+			mt.Run(tc.name, func(mt *mtest.T) {
+				bucket, err := gridfs.NewBucket(mt.DB, options.GridFSBucket().SetChunkSizeBytes(chunkSize))
+				assert.Nil(mt, err, "NewBucket error: %v", err)
+
+				ustream, err := bucket.OpenUploadStream("foo")
+				assert.Nil(mt, err, "OpenUploadStream error: %v", err)
+
+				id := ustream.FileID
+				_, err = ustream.Write(data)
+				assert.Nil(mt, err, "Write error: %v", err)
+				err = ustream.Close()
+				assert.Nil(mt, err, "Close error: %v", err)
+
+				dstream, _ := bucket.OpenDownloadStream(id)
```

Comment:
Is the error intentionally ignored here?

Reply:
Unintentional, fixed.